power_prediction/preprocess_data.py

```diff
@@ -11,77 +30,141 @@ def convert_date(df):
     df['Weekday'] = df['Date_format'].dt.weekday + 1
     pass
 
-
-def combine_power_consumption(df):
-    df['Power_consumption'] = df['Value_NE5'] + df['Value_NE7']
-
-
-def fill_nan_values(df):
-    columns = ['Hr [%Hr]', 'T [°C]', 'WD [°]', 'WVs [m/s]', 'WVv [m/s]', 'p [hPa]', 'RainDur [min]', 'StrGlo [W/m2]']
-    for column in columns:
-        nan_rows = df[df[[column]].isna().any(axis=1)]
-        mean_values = (df.groupby(['Month', 'Day', 'Hour'])[column].mean().to_dict())
-
-        for index, row in nan_rows.iterrows():
-            month = row['Month']
-            day = row['Day']
-            hour = row['Hour']
-            value = mean_values.get((month, day, hour))
-            df.at[index, column] = value
+def combine_power_consumption(df) -> None:
+    """Combines the power consumption columns."""
+    df['NE_tot'] = df['Value_NE5'] + df['Value_NE7']
     pass
 
+def clean_time_sync(df) -> pd.DataFrame:
+    """Removes the last rows."""
+    df_time = df.drop(df.index[85512:85525])
```

Comment:
Seems very hard coded. Could you just return something like:

```
return df[df["NE_tot"] == pd.nan && df["Date"] > pd.to_datetime("2024-xx-xx")]
```

You would surely have to try this out!